StripeConnect/StripeConnect/Source/Internal/Webview/MessageSenders/CallSetterWithSerializableValueSender.swift

```diff
@@ -0,0 +1,17 @@
+//
+//  CallSetterWithSerializableValueSender.swift
+//  StripeConnect
+//
+//  Created by Chris Mays on 8/7/24.
+//
+
+import Foundation
+
+struct CallSetterWithSerializableValueSender: MessageSender {
+    struct Message: Codable, Equatable {
+        let setter: String
+        let value: String
+    }
+    let name: String = "callSetterWithSerializableValue"
+    let message: Message
+}
```

Comment:
🚫 **Blocker:** The value could potentially be a different type. For example, `setCollectionOptions` on the onboarding component will be a complex type ([docs](https://docs.stripe.com/connect/supported-embedded-components/account-onboarding#create-an-account-session)). 

Looking at our internal docs, it looks like it's implied that `value` should be a serialized JSON-encoded string – so we might want to verify this is accurate (it would be easier if it didn't have to be serialized). If it does need to be serialized into a String, I think the best way to do it might be to implement custom encoding on the `Message` type

```suggestion
struct CallSetterWithSerializableValueSender<Value: Encodable>: MessageSender {
    struct Message: Encodable, Equatable {
        let setter: String
        let value: Value
        
        enum CodingKeys: String, CodingKey {
            case setter
            case value
        }
        
        func encode(to: any Encoder) throws {
            var container = encoder.container(keyedBy: CodingKeys.self)
            try container.encode(setter, forKey: .setter)
            let serializedValue = // encode value to JSON
            try container.encode(serializedValue, forKey: .value)
        }
    }
    
    let name: String = "callSetterWithSerializableValue"
    let message: Message
}
```

Reply:
FYI - it's okay to address this in a followup PR separately if you think that's appropriate